src/encryption/symmetric/aes/README.md

```diff
@@ -0,0 +1,123 @@
+# AES encryption
+
+[Advanced Encryption Standard][aes] (AES) is a symmetric-key algorithm and is a variant of the Rijndael (pronounced 'rain-dull') block cipher. AES supersedes the [Data Encryption Standard][des] (DES).
+
+## Overview
+
+AES supports a block size of 128 bits, and three different key lengths: 128, 192, and 256 bits. It manipulates _bytes_ instead of _individual bits_ or _64-bit words_, and views a 16-byte plaintext as a 2D column-major grid of bytes.
+
+AES uses [Substitution-Permutation Network (SPN)][spn] which makes use of two main properties: _confusion_ and _diffusion_. Confusion means that the input undergoes complex transformations, and diffusion means that these transformations depend equally on all bits of the input.
+
+Unlike DES, it does not use a Feistel network, and most AES calculations are done in a particular finite field.
+
+
+## Algorithm 
+
+The core encryption algorithm consists of the following routines:
+- [KeyExpansion](#KeyExpansion)
+- [AddRoundKey](#AddRoundKey)
+- [SubBytes](#SubBytes)
+- [ShiftRows](#ShiftRows)
+- [MixColumns](#MixColumns)
+
+For decryption, we take the inverses of these routines:
+
+TODO
```

Comment:
Complete or add more detail.

Reply:
Ah yeah, was thinking to leave decryption for a followup PR - apologies for not being clear about that, but I could also complete it within the same PR (if that's preferable)

Reply:
I'm fine either way

Reply:
Filed an issue for decryption: https://github.com/pluto/ronkathon/issues/119